packages/transactions/src/new-compile-transaction.ts

```diff
@@ -21,7 +39,19 @@ export function compileTransaction(transactionMessage: CompilableTransactionMess
         signatures[signerAddress] = null;
     }
 
-    const transaction: NewTransaction = {
+    let lifetimeConstraint: TransactionWithLifetime['lifetimeConstraint'];
+    if ('blockhash' in transactionMessage.lifetimeConstraint) {
+        lifetimeConstraint = {
+            blockhash: transactionMessage.lifetimeConstraint.blockhash,
+        };
+    } else {
+        lifetimeConstraint = {
+            nonce: transactionMessage.lifetimeConstraint.nonce,
+        };
+    }
```

Comment:
Since you're already doing this check in two spots, and we could introduce more, wdyt about following our library's pattern like `assertIsDurableNonceTransaction` and rolling one of these?

```ts
assertIsBlockhashLifetime(lifetime: T): asserts lifetime is BlockhashLifetime;
```

Reply:
I've just rolled in our existing `isTransactionMessageWithBlockhashLifetime` instead of doing the ad-hoc check. I think that makes the most sense since we're dealing with `CompilableTransactionMessage` anyway.